src/core/Akka.TestKit/TestKitBase_ExpectAllWithPredicates.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.TestKit.Internal;
+
+namespace Akka.TestKit;
+
+public abstract partial class TestKitBase
+{
+    public IReadOnlyCollection<object> ExpectMsgAllOfMatchingPredicates(params PredicateInfo[] predicates)
+    {
+        using var cts = new CancellationTokenSource(RemainingOrDefault);
+        return ExpectMsgAllOfMatchingPredicates(predicates, cts.Token);
+    }
+
+    public IReadOnlyCollection<object> ExpectMsgAllOfMatchingPredicates(IReadOnlyCollection<PredicateInfo> predicates,
+        CancellationToken cancellationToken)
+    {
+        return ExpectMsgAllOfMatchingPredicatesAsync(predicates, cancellationToken)
+            .ToListAsync(cancellationToken)
+            .ConfigureAwait(false).GetAwaiter().GetResult();
+    }
+
+    public async IAsyncEnumerable<object> ExpectMsgAllOfMatchingPredicatesAsync(
+        IReadOnlyCollection<PredicateInfo> predicates,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var enumerable = InternalExpectMsgAllOfMatchingPredicatesAsync(new TimeSpan(0, 0, 60), predicates,
+                cancellationToken: cancellationToken)
+            .ConfigureAwait(false).WithCancellation(cancellationToken);
+        await foreach (var item in enumerable)
```

Comment:
Can't we return the result of `InternalExpectMsgAllOfMatchingPredicateAsync` and avoid re-enumerating it?

Reply:
I think it is fine to just return the result instead, as long as we keep the `.ConfigureAwait()` code.

Reply:
Actually, nevermind, this code is good as is, its not possible to return an `IAsyncEnumerable` with `.ConfigureAwait()`